File: app/src/main/java/com/devanshisukhija/sicsrattendance/Controller/DatabaseHelper.java
package com.devanshisukhija.sicsrattendance.Controller;

import com.google.firebase.database.FirebaseDatabase;

/**
 * Created by devanshi on 22/02/18.
 */

public class DatabaseHelper {
    private static boolean persistenceEnable = false;
    private static FirebaseDatabase mDatabase;


    public static boolean isPersistenceEnable(){
        return persistenceEnable;
    }
    public static FirebaseDatabase getInstance() {
        if (mDatabase == null) {
            mDatabase = FirebaseDatabase.getInstance();
            if(persistenceEnable==true) {
                mDatabase.setPersistenceEnabled(true);
            }
        }

        return mDatabase;
    }
}

File: app/src/main/java/com/devanshisukhija/sicsrattendance/Utility/DatabaseHelper.kt
package com.devanshisukhija.sicsrattendance.Utility

import com.google.firebase.database.FirebaseDatabase

/**
 * Created by devanshi on 01/03/18.
 */

object DatabaseHelper {
    var isPersistenceEnable = false
    private var mDatabase: FirebaseDatabase? = FirebaseDatabase.getInstance()
    val instance: FirebaseDatabase
        get() {
            if (mDatabase == null) {
                mDatabase = FirebaseDatabase.getInstance()
                if (isPersistenceEnable == true) {
                    mDatabase!!.setPersistenceEnabled(true)
                }
            }

            return mDatabase!!
        }
}
